nixos/modules/services/web-apps/librechat.nix

```
@@ -0,0 +1,232 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librechat;
+  format = pkgs.formats.yaml { };
+  configFile = format.generate "librechat.yaml" cfg.settings;
+  exportCredentials = n: _: ''export ${n}="$(${pkgs.systemd}/bin/systemd-creds cat ${n}_FILE)"'';
+  exportAllCredentials = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList exportCredentials vars);
+  getLoadCredentialList = lib.mapAttrsToList (n: v: "${n}_FILE:${v}") cfg.credentials;
+in
+{
+  options.services.librechat = {
+    enable = lib.mkEnableOption "the LibreChat server";
+
+    package = lib.mkPackageOption pkgs "librechat" { };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Whether to open the port in the firewall.
+      '';
+    };
+
+    dataDir = lib.mkOption {
+      type = lib.types.path;
+      default = "/var/lib/librechat";
+      example = "/persist/librechat";
+      description = "Absolute path for where the LibreChat server will use as its data directory to store logs, user uploads, and generated images.";
+    };
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "librechat";
+      example = "alice";
+      description = "The user to run the service as.";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "librechat";
+      example = "users";
+      description = "The group to run the service as.";
+    };
+
+    credentials = lib.mkOption {
+      type = lib.types.attrsOf lib.types.path;
+      default = { };
+      example = {
+        CREDS_KEY = /run/secrets/creds_key;
```

```suggestion
      type = lib.types.attrsOf (oneOf [
            str
            (coercedTo path toString str)
      ]);
      default = { };
      example = {
        CREDS_KEY = "/run/secrets/creds_key";
```

I'm not even certain we *should* allow paths, because they copy the credentials to the nix store, which is obviously not great.

---

Oh, I misunderstood what you meant. My bad sorry LMFAO